### DISCUSSION TITLE
ban

### DISCUSSION THREAD

**Discussion by comdar16** [Pryaxis/TShock#2477] [Q&A] [answered]
How can i ban offline user?

because it always says something invalid

**Comment by Arthri** [comment, marked as answer]
What command are you running? And who are you trying to ban?

If the user has an account, do `/accountinfo <Account Name>` first. That command will give you information about the user. Then you can use `/ban -a <Account Name> -n <Account Name> -ip <IP Address>` to ban them. Note, you can't do UUID bans at the moment without directly accessing the database

If the user doesn't have an account, then you will have to browse the logs to find their IP address. After that you can use `/ban -n <Player Name> -ip <IP Address>`. TShock offline bans for non-registered user has always been hard since it doesn't save much info about them

I understand that it may be tedious, but you're not alone, you can upvote #2427

**Reply by comdar16** [reply]
kinda weird, it would be better if you can just execute /ban <username>
